virtual_rainforest/main.py

```diff
@@ -227,12 +308,26 @@ def vr_run(
         )
 
     # Setup the timing loop
+    time_index = 0
     while current_time < end_time:
         current_time += update_interval
 
         # Run update() method for every model
         for mod_nm in models_cfd:
-            models_cfd[mod_nm].update()
+            models_cfd[mod_nm].update(time_index)
+
+        # With updates complete increment the time_index
+        time_index += 1
+
+        # Append updated data to the continuous data file
+        if config["core"]["data_output_options"]["save_continuous_data"]:
+            output_current_state(
+                data, models_cfd, config["core"]["data_output_options"], time_index
+            )
```

Comment:
I think this should return the Path of the created file and store that path in a list, which can then be used as an input to `merge_continuous_data_files`. So before the timing loop starts: `continuous_data_files = []` and then:

```python
continuous_data_files.append(
    output_current_state(
                data, models_cfd, config["core"]["data_output_options"], time_index
            )
)
```
This way, the merge operation is fed an explicit list of files to handle. I would also be tempted to separate the delete operation from the merge operation.